src/rendering/SceneManager.cc

```diff
@@ -258,16 +258,11 @@ rendering::VisualPtr SceneManager::CreateVisual(Entity _id,
     /// In general, this can be used to store any local transforms between the
     /// parent Visual and geometry.
     rendering::VisualPtr geomVis;
-    if (localPose != math::Pose3d::Zero)
-    {
-      geomVis = this->dataPtr->scene->CreateVisual(name + "_geom");
-      geomVis->SetUserData("gazebo-entity", static_cast<int>(_id));
```

Comment:
Is there a downside to keeping the user data in? I'm not sure if there are other GUI plugins relying on them, and these plugins don't have automated tests, so this may be breaking something that will be tough to track down later. I'm more concerned about the entity data than the pause update.

Reply:
I don't know for sure, but my reasoning was that the user data was needed here because we had `visualVis = geomVis`, i.e, `visualVis` was being replaced. Now, `visualVis` has the user data and we are just adding a `geomVis` has a child node.

Reply:
Got it. One possible problem of having the data there would be that the wrong visual is returned when trying to get the visual with the given entity ID.